### PR TITLE
Document Broker Delivery Options

### LINKED ISSUE #2043: document broker delivery options
## What is changing? (Please include as many details as possible.)
The Broker now supports `spec.delivery`.

## How will this impact our users?
They will be able to control what to do (retries, deadlettersink) when event delivery fails.

## In what release will this take happen (to the best of your knowledge)?
Ex. v0.2-v0.3
0.12.0

## Context
Link to associated PRs or issues from other repos here.

1. https://github.com/knative/eventing/pull/2305
1.
1.

### PR DESCRIPTION
**Pull request by pierDipi**
Fixes #2043 

## Proposed Changes 

- Document Broker Delivery Options

### CODE REVIEW

File: docs/eventing/event-delivery.md [hunk 1]
```diff
@@ -63,7 +96,17 @@ The table below summarizes what delivery parameters are supported for each chann
...
+
+## Broker Support
+
+The table below summarizes what delivery parameters are supported for each Broker implementation.
+
+| Broker Class | Supported Delivery Parameters |
+| - | - |
+| Kafka | `deadLetterSink`, `retry`, `backoffPolicy`, `backoffDelay` |
+| MTChannelBasedBroker | depends on the underlying channel |
+| RabbitMQBroker | `deadLetterSink`, `retry`, `backoffPolicy`, `backoffDelay` |
```
Comment (pierDipi) [inline docs/eventing/event-delivery.md:112]: Is this correct?

/cc vaikas n3wscott

Comment (vaikas) [reply docs/eventing/event-delivery.md:112]: Yes! Thank you!!!

File: docs/eventing/event-delivery.md [hunk 2]
```diff
@@ -63,7 +96,17 @@ The table below summarizes what delivery parameters are supported for each chann
...
 | Kafka | `deadLetterSink`, `retry`, `backoffPolicy`, `backoffDelay` |
 | Natss | none |
-| GCP PubSub | none |
+
+## Broker Support
+
+The table below summarizes what delivery parameters are supported for each Broker implementation.
+
+| Broker Class | Supported Delivery Parameters |
+| - | - |
```
Comment (grantr) [inline docs/eventing/event-delivery.md:109]: @pierDipi could you please add these lines for GCP Broker and its footnotes? Thank you!

```
| googlecloud | `deadLetterSink` [^1], `retry`, `backoffPolicy`, `backoffDelay` [^2] |

[^1]: deadLetterSink must be a GCP Pub/Sub topic uri:
    ```yaml
    deadLetterSink:
      uri: pubsub://dead-letter-topic
    ```

    Please see the
    [config-br-delivery](https://github.com/google/knative-gcp/blob/master/config/core/configmaps/br-delivery.yaml)
    ConfigMap for a complete example.

[^2]: The googlecloud broker only supports the `exponential` backoffPolicy.
```

**Comment by pierDipi** [comment]
I need to add GCP Broker.
/hold

File: docs/eventing/event-delivery.md (hunk 2 above)
Comment (pierDipi) [reply docs/eventing/event-delivery.md:109]: Done https://github.com/knative/docs/pull/2938/commits/3a11af50615c9a08caa4224923451fb410137e80

/cc @grantr

Comment (grantr) [reply docs/eventing/event-delivery.md:109]: Thanks @pierDipi! Looks like github doesn't support the footnote syntax I used. Oh well. Feel free to reformat however you want.

**Review by grantr — APPROVED** [review]
/lgtm

**Pull request merged**